ext/opentelemetry-ext-pymongo/src/opentelemetry/ext/pymongo/__init__.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The opentelemetry-ext-pymongo package allows tracing commands made by the
+pymongo library.
+"""
+
+from pymongo import monitoring
+
+from opentelemetry.trace import SpanKind
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+DATABASE_TYPE = "mongodb"
+COMMAND_ATTRIBUTES = ["filter", "sort", "skip", "limit", "pipeline"]
+
+
+def trace_integration(tracer=None):
+    """Integrate with pymongo to trace it using event listener.
+       https://api.mongodb.com/python/current/api/pymongo/monitoring.html
+    """
+
+    monitoring.register(CommandTracer(tracer))
+
+
+class CommandTracer(monitoring.CommandListener):
+    def __init__(self, tracer):
+        if tracer is None:
+            raise ValueError("The tracer is not provided.")
+        self._tracer = tracer
+        self._span = None
```

Comment:
It seems to me that the same CommandTracer instance is used for all MongoClient objects. Thus, it seems that the cure of using an instance attribute `_span` is worse than the disease of using `tracer.get_active_span()`.

I think we could improve the situation by storing the span in its own context variable.

I guess there might be scenarios with recursive MongoDB ivocations (or might there really?) where even that is not enough, then you would need to instead have a dictionary instance attribute indexed by `event.request_id` (or maybe a tuple of request_id and connection_id, the pymongo docs are not clear on within which scopes the IDs are unique).

Actually that second approach sounds even a bit cleaner to me.

Reply:
Added logic you suggested, looking at pymongo OpenCensus integration it just rely on current_span, it feels like a potential bug I will follow up as well. Thanks a lot for reviewing